lib/src/markdown_processor.dart

```diff
@@ -130,58 +112,38 @@ MatchingLinkResult _getMatchingLinkElement(
 }
 
 MatchingLinkResult _findRefElementInLibrary(String codeRef, ModelElement element, List<CommentReference> commentRefs) {
-  final package = element.library.package;
+  final Package package = element.library.package;
   final Map<String, ModelElement> result = {};
-  package.libraries.forEach((library) {
-    final List<ModelElement> modelElements = []
-      ..addAll(library.allClasses)
-      ..addAll(library.constants)
-      ..addAll(library.enums)
-      ..addAll(library.functions)
-      ..addAll(library.properties)
-      ..addAll(library.typedefs);
-
-    library.allClasses.forEach((c) {
-      modelElements.addAll(c.allInstanceMethods);
-      modelElements.addAll(c.allInstanceProperties);
-      modelElements.addAll(c.allOperators);
-      modelElements.addAll(c.staticMethods);
-      modelElements.addAll(c.staticProperties);
-    });
-
-    modelElements.forEach((modelElement) {
-      if (codeRef == modelElement.name || codeRef == modelElement.fullyQualifiedName) {
-        result[modelElement.fullyQualifiedName] = modelElement;
-      }
-    });
+  package.allModelElements.forEach((modelElement) {
```

Comment:
Last comment, I promise :)

If we use a Map here, in the event we have multiple matches, the entry in the map could be overridden. So that would hide the fact that the reference is not unique. Perhaps use a List instead?

And we'll be iterating over a lot of elements - I tend to avoid using a closure for that; a for loop will be more performant.

You might do something thuswise:

```dart
List<ModelElement> results = [];

for (ModelElement element in package.allModelElements) {
  if (codeRef.contains('.')) {
    // It's a fully qualified reference.
    if (element.fullyQualifiedName == codeRef) {
      results.add(element);
      break;
    }
  } else {
    if (element.name == codeRef) {
      results.add(element);
    }
  }
}
```